flink-dist/src/main/flink-bin/bin/config.sh

```diff
@@ -429,6 +430,11 @@ if [ -z "${FLINK_TM_NET_BUF_MAX}" -o "${FLINK_TM_NET_BUF_MAX}" = "-1" ]; then
     FLINK_TM_NET_BUF_MAX=$(parseBytes ${FLINK_TM_NET_BUF_MAX})
 fi
 
+# Define FLINK_TM_MAX_OFFHEAP_SIZE if it is not already set
+if [ -z "${FLINK_TM_MAX_OFFHEAP_SIZE}" ]; then
+    # default: Long.MAX_VALUE in TB
+    FLINK_TM_MAX_OFFHEAP_SIZE=$(readFromConfig ${KEY_TASKM_MAX_OFFHEAP_SIZE} "8388607T" "${YAML_CONF}")
```

Comment:
Should the condition be `if [ -z "${FLINK_TM_MAX_OFFHEAP_SIZE}" -o "${FLINK_TM_MAX_OFFHEAP_SIZE}" = "-1" ]`?

Reply:
I am not sure. In what case would ${FLINK_TM_MAX_OFFHEAP_SIZE} be equal to -1?

Reply:
Maybe it is set to -1 by user incorrectly. You can find other configurations like `FLINK_TM_NET_BUF_MAX` use this condition too.

Reply:
Resolved, thanks!